machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +54,20 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
+	// GuestAccelerators is a list of GPUs to be attached to the VM.
+	// +optional
+	GuestAccelerators []*GCPAcceleratorConfig `json:"guestAccelerators,omitempty"`
 	// Preemptible indicates if created instance is preemptible
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance Determines the behavior when a maintenance event occurs that might cause the instance to reboot (default MIGRATE).
+	// This is required to be set to TERMINATE if you want to provision machine with attached GPUs.
+	// +optional
+	OnHostMaintenance string `json:"onHostMaintenance,omitempty"`
```

Comment:
Consider defining a type and named constants for the allowed values.  

Are "MIGRATE" and "TERMINATE" the only allowed values?  

Can you elaborate on what these values mean?  I can only guess that "MIGRATE" could mean the cloud platform would do some kind of live migration of the VM, whereas "TERMINATE" could mean the cloud platform would do a regular OS shutdown, forcing the eviction of pods.  

```suggestion
	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
	// Allowed values are "MIGRATE" or "TERMINATE".  The default value is "MIGRATE".
	// "MIGRATE" means [...].  "TERMINATE" means [...].  
	// This is required to be set to "TERMINATE" if the machine has attached accelerators.
	// +optional
	OnHostMaintenance string `json:"onHostMaintenance,omitempty"`
```

Reply:
This as well is set as string following google compute api. I see that for example in google's .NET api package they have it set as enum. However in golang package it is simply a string. We could possibly change to its own type, however it doesnt seem necessary to me, considering we would be diverging from compute api convntion.
 I can add an extra comment concerning MIGRATE and TERMINATE meaning.